safe/gui/tools/field_mapping_dialog.py

```diff
@@ -0,0 +1,284 @@
+
+        self.field_mapping_widget = None
+        self.main_stacked_widget.setCurrentIndex(1)
+
+        # Input
+        self.layer_input_layout.addWidget(self.layer_label)
+        self.layer_input_layout.addWidget(self.layer_combo_box)
+
+        self.header_label = QLabel()
+        self.main_layout.addWidget(self.header_label)
+        self.main_layout.addLayout(self.layer_input_layout)
+
+        # Signal
+        self.layer_combo_box.layerChanged.connect(self.set_layer)
+
+        if self.layer_combo_box.currentLayer():
+            self.set_layer(self.layer_combo_box.currentLayer())
+
+        # Set up things for context help
+        self.help_button = self.button_box.button(QDialogButtonBox.Help)
+        # Allow toggling the help button
+        self.help_button.setCheckable(True)
+        self.help_button.toggled.connect(self.help_toggled)
+
+        # Set up things for ok button
+        self.ok_button = self.button_box.button(QDialogButtonBox.Ok)
+        self.ok_button.clicked.connect(self.accept)
+
+        # Set up things for cancel button
+        self.cancel_button = self.button_box.button(QDialogButtonBox.Cancel)
+        self.cancel_button.clicked.connect(self.reject)
+
+    def set_layer(self, layer=None, keywords=None):
+        """Set layer and update UI accordingly.
+
+        :param layer: A QgsVectorLayer.
+        :type layer: QgsVectorLayer
+
+        :param keywords: Keywords for the layer.
+        :type keywords: dict, None
+        """
+        if self.field_mapping_widget is not None:
+            self.field_mapping_widget.setParent(None)
+            self.field_mapping_widget.close()
+            self.field_mapping_widget.deleteLater()
+            self.main_layout.removeWidget(self.field_mapping_widget)
+
+        if layer:
+            self.layer = layer
+        else:
+            self.layer = self.layer_combo_box.currentLayer()
+        if not self.layer:
+            return
+
+        if keywords is not None:
+            self.metadata = keywords
+        else:
+            # Always read from metadata file.
+            try:
+                self.metadata = self.keyword_io.read_keywords(self.layer)
+            except (
+                NoKeywordsFoundError,
+                KeywordNotFoundError,
+                MetadataReadError) as e:
+                raise e
+        if 'inasafe_default_values' not in self.metadata:
+            self.metadata['inasafe_default_values'] = {}
+        if 'inasafe_fields' not in self.metadata:
+            self.metadata['inasafe_fields'] = {}
+        self.field_mapping_widget = FieldMappingWidget(
+            parent=self, iface=self.iface)
+        self.field_mapping_widget.set_layer(self.layer, self.metadata)
+        self.field_mapping_widget.show()
+        self.main_layout.addWidget(self.field_mapping_widget)
+
+        # Set header label
+        group_names = [self.field_mapping_widget.tabText(i) for i in
+             range(self.field_mapping_widget.count())]
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->